pxfish/code.py

```diff
@@ -5,39 +5,86 @@
 import os
 
 
-def write(path, file_name, code_object):
+def write(*, path, file_name, code_object):
     """
-    Writes the aquarium code object to the given path.
+    Writes the content of the aquarium code object to the given path.
 
     Arguments:
       path (string): the path of the file to be written
       file_name (string): the name of the file to be written
-      code_object (Code): the code object
+      code_object (Code): the code object to be written
     """
     file_path = os.path.join(path, file_name)
     with open(file_path, 'w') as file:
         file.write(code_object.content)
 
 
-def create_code_objects(aq, component_names):
+def create_code_objects(*, session, component_names):
     """
     Creates code objects for each named component.
 
     Arguments:
-        aq (Session Object): Aquarium session object
+        session (Session Object): Aquarium session object
         component_names (List): names of code components
+
+    Returns:
+        code_objects (Dictionary): Dictionary with component names as keys
+        and retrieved Code objects as values.
     """
     code_objects = {}
+
     for name in component_names:
-        code_objects[name] = aq.Code.new(name=name, content='')
+        default_content = add_default_content(name)
+        code_objects[name] = session.Code.new(
+                                    name=name, content=default_content)
     return code_objects
 
 
+def add_default_content(name):
+    """
+    Retrieves default content for code files.
+
+    Arguments:
+        name (String): name of file whose contents to retrieve
+    """
+    path = os.path.normpath("protocol_templates")
```

Comment:
The version on my machine has this path as `/script/protocol_templates`.
The Dockerfiles puts the pfish code into `/script` but then sets the working directory to `/wd` which is used by the wrapper script to mount the user working directory.

So, does this actually work with wrapper script and docker image?